NodeJS/__integration__/AddressBookClient.it.ts

```diff
@@ -0,0 +1,111 @@
+    expect(addressBookPayload.contacts.length).toBeGreaterThan(0)
+  }, 3000)
+
+  test('searchContacts w/o query params should return all active contacts', async () => {
+    const addressBookSearchResponse = await addressBookClient.searchContacts()
+    console.log('GET searchContacts ', addressBookSearchResponse)
+    expect(addressBookSearchResponse).toBeDefined()
+    expect(addressBookSearchResponse.content.length).toBeGreaterThan(0)
+    expect(addressBookSearchResponse.number).toEqual(0)
+    expect(addressBookSearchResponse.totalPages).toBeGreaterThan(0)
+  }, 3000)
+
+  test('searchContacts of ACO type should return active ACO contacts', async () => {
+    const searchRequest = { type: 'ACO' }
+    const addressBookSearchResponse = await addressBookClient.searchContacts(
+      searchRequest
+    )
+    console.log('GET searchContacts ', addressBookSearchResponse)
+    expect(addressBookSearchResponse).toBeDefined()
+    expect(addressBookSearchResponse.content.length).toBeGreaterThan(0)
+    expect(addressBookSearchResponse.number).toEqual(0)
+    expect(addressBookSearchResponse.totalPages).toBeGreaterThan(0)
+    for (const contact of addressBookSearchResponse.content) {
+      expect(contact.type).toEqual('ACO')
+    }
+  }, 3000)
+
+  test('searchContacts using query should return corresponding contacts', async () => {
+    const searchRequest = { query: 'agent' }
+    const addressBookSearchResponse = await addressBookClient.searchContacts(
+      searchRequest
+    )
+    console.log('GET searchContacts ', addressBookSearchResponse)
+    expect(addressBookSearchResponse).toBeDefined()
+    expect(addressBookSearchResponse.content.length).toBeGreaterThan(0)
+    expect(addressBookSearchResponse.number).toEqual(0)
+    expect(addressBookSearchResponse.totalPages).toBeGreaterThan(0)
+    for (const contact of addressBookSearchResponse.content) {
+      expect(contact.name).toContain('agent')
+      console.log(contact.name)
+    }
+  }, 3000)
+
+  test('searchContacts sorted by should return all active contacts in expected order', async () => {
+    const searchRequest = { orderBy: 'name', orderDirection: 'DESC' }
+    const addressBookSearchResponse = await addressBookClient.searchContacts(
+      searchRequest
+    )
+    console.log('GET searchContacts ', addressBookSearchResponse)
+    expect(addressBookSearchResponse).toBeDefined()
+    expect(addressBookSearchResponse.content.length).toBeGreaterThan(0)
+    expect(addressBookSearchResponse.number).toEqual(0)
+    expect(addressBookSearchResponse.totalPages).toBeGreaterThan(0)
+    // Collect all contact names and sort in expected order
+    const names: string[] = []
+    let i = 0
+    for (const contact of addressBookSearchResponse.content) {
+      names[i++] = contact.name
+    }
+    names.sort()
+    names.reverse()
+    let j = 0
+    for (const contact of addressBookSearchResponse.content) {
+      expect(contact.name).toEqual(names[j++])
+    }
+  }, 3000)
```

Comment:
This test doesn't look right. Your expecting the result to be ordered, but then you're sorting the result and comparing it against itself? I'm quite confused...

Reply:
searchRequest is { orderBy: 'name', orderDirection: 'DESC' }
in the 1st loop, I'm storing names in the array. then sorting in reverse order (as it expected to be returned by contact-service).
in the 2nd loop, I'm comparing the actual contact name with the expected contact name from my sorted array.

Reply:
refactored to make it shorter and nicer

Reply:
yup yup, this is much clearer to me, thanks!